### PR TITLE
Factors trigger

### PR DESCRIPTION
**Pull request by lann**
This turned into a bit of a mega-PR, but I have split out some smaller commits, including:
- Add `SpinFilesMounter`
- Add intermediate `FactorsExecutorApp` type

Note that while this mostly a straight port, I have also implemented 2 breaking changes for Spin 3.0:

- Drop support for executing WAGI modules
  - still need to fix up detection and errors for uncomponentizable modules
- Drop the HTTP trigger `base` field 
  - the default value of `/` is still permitted with a warning

I've left the existing `spin-trigger` and `spin-trigger-http` crates for now as there is still some functionality that needs to be ported to factors:

- Outbound HTTP features that were implemented in `spin-trigger-http`: client TLS, `self` and service chained requests
- `spin-trigger-http` benchmarks

### CODE REVIEW

File: crates/factors-executor/src/lib.rs [hunk 1]
```diff
@@ -10,85 +10,137 @@ use spin_factors::{AsInstanceState, ConfiguredApp, RuntimeFactors, RuntimeFactor
...
-            core_engine,
+        Ok(FactorsConfiguredApp {
+            executor: self,
             configured_app,
             component_instance_pres,
         })
     }
+}
+
+pub trait ExecutorHooks<T: RuntimeFactors, U>: Send {
```
[the remark this answers is not in the document]
Comment (lann) [reply crates/factors-executor/src/lib.rs:78]: These are replacing `TriggerHooks` so there will be a bunch of examples once they are all ported.

File: crates/trigger-http2/src/handler.rs [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+use spin_http::routes::RouteMatch;
+use spin_world::v1::http_types;
+use tokio::{sync::oneshot, task};
+use tracing::{instrument, Instrument, Level};
+use wasmtime_wasi_http::{body::HyperIncomingBody as Body, proxy::Proxy, WasiHttpView};
+
+use crate::{server::HttpExecutor, Store, TriggerInstanceBuilder};
+
+#[derive(Clone)]
+pub struct HttpHandlerExecutor {
```
Comment (rylev) [inline crates/trigger-http2/src/handler.rs:21]: It'd be good to document what this is - an `HttpExector` that can execute wasi and spin http incoming requests.

**Review by rylev — APPROVED** [review]
This is when things start to get a bit scary - the trigger2 crate has a lot of overlap with the existing trigger crate, and it's hard to see where things differ. I think we're at the point that I'd like to see integration of factors happened as quickly as possible so we stop living in a world where things can easily diverge.

**Comment by lann** [comment]
> This is when things start to get a bit scary - the trigger2 crate has a lot of overlap with the existing trigger crate, and it's hard to see where things differ. I think we're at the point that I'd like to see integration of factors happened as quickly as possible so we stop living in a world where things can easily diverge.

Agreed. I've been struggling with a more principled refactor here and at this point I just want to get something merged to build on.

**Pull request merged**